repl-plots/src/main/scala/com/cibo/evilplot/displayPlot.scala

```diff
@@ -0,0 +1,144 @@
+ *    may be used to endorse or promote products derived from this software without
+ *    specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.cibo.evilplot
+
+import java.awt.{Graphics, Graphics2D}
+
+import com.cibo.evilplot.plot.{Plot}
+import javax.swing.{JFileChooser, JFrame, JPanel}
+import java.awt.event.{ActionEvent, ComponentAdapter, ComponentEvent}
+import java.io.File
+
+import com.cibo.evilplot.demo.DemoPlots.theme
+import com.cibo.evilplot.geometry.{Drawable, Extent}
+
+object displayPlot {
+
+  private class DrawablePanel extends JPanel {
+    var drawable: Option[Drawable] = None
+
+    def setDrawable(drawnPlot: Drawable): Unit = {
+      drawable = Some(drawnPlot)
+    }
+
+    override def paintComponent(g: Graphics): Unit = {
+      super.paintComponent(g)
+      val g2 = g.asInstanceOf[Graphics2D]
+      if (!(drawable == None)) {
+        g2.drawImage(drawable.get.asBufferedImage, -30, 0, this)
+
+      }
+    }
+  }
+
+  private class DrawableFrame extends JFrame {
+
+    import javax.swing.JMenuBar
+    import javax.swing.JMenuItem
+
+    var drawable: Option[Drawable] = None
+    var plot: Option[Plot] = None
+    val panel: DrawablePanel = new DrawablePanel()
+
+    private def createMenuBar(): Unit = {
+      val menubar = new JMenuBar
+      val save = new JMenuItem("Save")
+      save.addActionListener((event: ActionEvent) => {
+        def save(event: ActionEvent) = {
+          val selectFile = new JFileChooser()
+          selectFile.setCurrentDirectory(new File("~"))
+          val savedFile: Int = selectFile.showSaveDialog(this)
+          if (savedFile == JFileChooser.APPROVE_OPTION) {
+            val file = selectFile.getSelectedFile
+            savePlot(file)
+          }
+        }
+        save(event)
+      })
+      menubar.add(save)
+      setJMenuBar(menubar)
+    }
+
+    private def init(): Unit = {
+      setTitle("Plot")
+      if (!drawable.isEmpty) {
+        setSize(drawable.get.extent.width.toInt * 2, drawable.get.extent.height.toInt * 2 + 20)
+        panel.setDrawable(drawable.get.scaled(0.5, 0.5))
+      } else {
+        setSize(400, 420)
```

Comment:
perhaps the default size for plot is a better option here?

Reply:
The default size, even for the bufferedImage, is massive in swing for some reason. I'm keeping the extents to match the size of the window but everything has to be scaled down just to fit on the screen.